### ISSUE TITLE
Lookup: Page number for gird is showing placeholder instead of actual values #35

### ISSUE THREAD

**Issue by helloimYS** [infor-design/enterprise-ng#792]
**Describe the bug**
_Follow-up on issue [infor-design/enterprise-ng#790](https://github.com/infor-design/enterprise-ng/issues/790)_
After install ids-enterprise v4.27.2, the grid is fixed but the values of page number, results and cancel became placeholder.
![image](https://user-images.githubusercontent.com/63391710/79415950-1da5dd80-7fe1-11ea-9d0c-86f45ee64e96.png)

**To Reproduce**
1. Unzip the folder below, navigate into the project folder and run `npm install` to install all dependency and run `odin serve` to start the project.
[Sample.zip](https://github.com/infor-design/enterprise-ng/files/4484947/Sample.zip)
2. Open the lookup, page number is not showing correctly, the text for Results and Cancel are enclosed with **'[ ]'**.

**Expected behavior**
Page number should be show correspondingly, text of Results and Cancel should be showing exact value but not placeholder ([Result], [Cancel]).

**Version**
- ids-enterprise-ng: v5.0.0
- ids-enterprise: v4.27.2

**Platform**
 - OS Version: Windows 10
 - Browser Name: Chrome
 - Browser Version: 80.0.3987.163 (Official Build) (64-bit)

**Additional context**
The input field might not be shown on first launch, try to refresh the browser until the input field is show on screen.

**Comment by tmcconechy** [comment]
When the [squares] are shown it means the translations are not found. You can add these by calling `Locale.set`.

In the latest version of Ng we use the new soholocaleinitialize. See the quick start https://github.com/infor-design/enterprise-ng-quickstart/tree/master/src/app

If you really on NG 5.0 you wont have this so possible solutions are:

1) Call this line in app.component.ts to set the locale (seems to work in your demo app. But this can be doing an ajax call so you might have to wait for it to get the locale and load it.

```
import { Component } from '@angular/core';
import { CoreBase } from '@infor-up/m3-odin';

@Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.css']
})
export class AppComponent extends CoreBase {
   constructor() {
      super('AppComponent');
      Soho.Locale.set('en-US');
   }
}
```
2) import the en-US.js script from node_modules/ids-enterprise/cultures in the head

**Comment by tmcconechy** [comment]
Hope that helps :)

**Comment by helloimYS** [comment]
Thank you for your prompt response @tmcconechy !

But I have goes into another issue. I upgraded my project to the latest Ng version in order to use the SohoLocaleinitialize. But when I  import `SohoLocaleInitializerModule` to `app.module.ts` it prompt the following issue:
![image](https://user-images.githubusercontent.com/63391710/79540921-7fd31100-80bb-11ea-845f-d343a31a37ca.png)
Could you please assist me on this issue?

Besides, may I ask how to import "en-US.js" script from node_modules/ids-enterprise/cultures? I've tried adding `<script src="../node_modules/ids-enterprise/dist/js/cultures/en-US.js"></script>` to `<head>` however it doesn't work :(

**To Reproduce**
Steps to reproduce the behavior:
1. Unzip the folder below, navigate into the project folder and run `npm install`  then `odin serve` to start the project.
[sample-latest-ng.zip](https://github.com/infor-design/enterprise-ng/files/4491553/sample-latest-ng.zip)
2. Inspect chrome and error show in console

**Version**
- ids-enterprise-ng: v7.0.0

**Comment by tmcconechy** [comment]
@helloimYS Looks like you just need to add the cultures (and maybe other stuff) to your assets in angular.json

See our sample apps:
https://github.com/infor-design/enterprise-ng/blob/master/angular.json#L26-L35
https://github.com/infor-design/enterprise-ng-quickstart/blob/master/angular.json#L22

**Comment by helloimYS** [comment]
Hi @tmcconechy 

It showing the same error as above although I've added the cultures to angular.json.
![image](https://user-images.githubusercontent.com/63391710/79740702-b36ba080-8332-11ea-9cae-2ca0fd4ae16c.png)

If I add `<script src="assets/ids-enterprise/js/cultures/en-US.js"></script>` to `<head>`, it fix the _TypeError_ error but it doesn't fix the errors throw by `sohoxi.js`.
![image](https://user-images.githubusercontent.com/63391710/79742069-dac36d00-8334-11ea-9396-a026213a09f8.png)

According to [this](https://github.com/infor-design/enterprise-ng-quickstart/blob/master/src/index.html) sample app, it doesn't require to manually add the `<script src="assets/ids-enterprise/js/cultures/en-US.js"></script>` to `<head>`.

**Comment by tmcconechy** [comment]
Your path looks a little different. Also the culturesPath in the soho-locale-initializer should match the output folder.

Give that a try, noting that the demo app is working https://github.com/infor-design/enterprise-ng-quickstart/blob/master/src/index.html

Please forget my comment about adding the script to head. Thats not the best way to do it if your planning on updating anyways. I have no example for that was just an idea.

**Comment by helloimYS** [comment]
Hi @tmcconechy 

Yes, I've try with both path and it still throwing the same error.
![image](https://user-images.githubusercontent.com/63391710/79821296-f70ceb80-83c0-11ea-8d28-a6d3ebebcba3.png)

The culturesPath is match with the output folder.
![image](https://user-images.githubusercontent.com/63391710/79821490-7ef2f580-83c1-11ea-89b2-515c493cafbb.png)

The error is throw by `await`, if I remove the 'await' keyword the error will gone but the message in `console.log` doesn't show. Maybe it's because the application start without waiting the locales to load. 
![image](https://user-images.githubusercontent.com/63391710/79822689-5fa99780-83c4-11ea-82df-e8fffa54ac53.png)

I did try on the quickstart demo app and yes it's working, I have cross check every files and folder however not sure why I'm facing this error in my project.

Kindly refer to the repository below for error reproduce.
https://github.com/helloimYS/Testing/tree/master/sample/sample-latest-ng

**Comment by helloimYS** [comment]
I re-launch my project in local just now and all the errors have gone, weird..
You're right after I change the input path and re-launch it the errors are solved.

Thank you @tmcconechy for your continued support :)

**Comment by tmcconechy** [comment]
OK, so you solved it? Maybe it worked because you cleaned the data out. One "trick" i use is to clean it like this to refresh everything out..
```
rm -rf node_modules
npm i
npm run build
npm run start
```

**Comment by helloimYS** [comment]
Yes it’s solve, but it only work fine in my local :(
When I run `odin build` to create a production folder and deploy it to M3, the application seem to be blocked by the `await` and cause the application view (interface) became blank.
![](https://user-images.githubusercontent.com/63391710/79856200-9c928000-83fe-11ea-83ca-ece99a6a9b19.png)

Do this mean `LocaleInitializer` doesn’t work on M3?

**Comment by helloimYS** [comment]
Yes it’s solve, but it only work fine in my local :(
When I run `odin build` to create a production folder and deploy it to M3, the application seem to be blocked by the `await` and cause the application view (interface) became blank.
![](https://user-images.githubusercontent.com/63391710/79856200-9c928000-83fe-11ea-83ca-ece99a6a9b19.png)

Do this mean `LocaleInitializer` doesn’t work on M3?
